rest/data/project_settings.go

```diff
@@ -410,12 +410,60 @@ func SaveProjectSettingsForSection(ctx context.Context, projectId string, change
 		modified, err = updateAliasesForSection(projectId, changes.Aliases, before.Aliases, section)
 		catcher.Add(err)
 	case model.ProjectPageNotificationsSection:
-		if err = SaveSubscriptions(projectId, changes.Subscriptions, true); err != nil {
+		subscriptionChanges := []restModel.APISubscription{}
+		for _, subscription := range changes.Subscriptions {
```

Comment:
I feel like we need some comment here explaining what the purpose of the loop is (or a helper function with a comment), because it's not that explicit how the redaction placeholder string interacts with the UI and saving the page for a new vs. existing subscription.

Can we mention in a comment that Evergreen has to allow a new subscription to be added as-is, but when a user saves the UI for an existing subscription after the value has been redacted, we have to ensure that the `<REDACTED>` values that it's using as placeholders don't end up overwriting the real secret value?